library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemAdapter.kt

```diff
@@ -4,40 +4,20 @@ import android.content.Context
 import android.view.ViewGroup
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.RecyclerView
-import com.getstream.sdk.chat.view.MessageListView
-import com.getstream.sdk.chat.view.MessageListView.AttachmentClickListener
 import com.getstream.sdk.chat.view.MessageListView.BubbleHelper
-import com.getstream.sdk.chat.view.MessageListView.GiphySendListener
-import com.getstream.sdk.chat.view.MessageListView.MessageClickListener
-import com.getstream.sdk.chat.view.MessageListView.MessageLongClickListener
-import com.getstream.sdk.chat.view.MessageListView.ReactionViewClickListener
-import com.getstream.sdk.chat.view.MessageListView.ReadStateClickListener
 import com.getstream.sdk.chat.view.MessageListViewStyle
 import io.getstream.chat.android.client.models.Channel
 
 class MessageListItemAdapter @JvmOverloads constructor(
     private var context: Context,
-    var channel: Channel? = null,
-    private var messageListItemList: List<MessageListItem> = emptyList(),
-    var viewHolderFactory: MessageViewHolderFactory = MessageViewHolderFactory()
+    val channel: Channel,
```

Comment:
I like it! Very nice to provide required dependencies via constant values in the constructor